### PR TITLE
api: lease comparison target

### PR DESCRIPTION
**Pull request by heyitsanthony**
It's useful to perform some atomic actions based on leases. For disconnected linearized reads, keys with ttls can't be easily leased since if the client is disconnected it doesn't know whether the ttl expires. Although there could be a Get in the leasing txn to later undo the leasing if the key turns out to have a ttl, this is inefficient and less reliable than gating it at the txn level.

I've omitted clientv3 support for the time being since calling the comparison function Lease to follow the current convention would conflict with the lease interface

### CODE REVIEW

File: etcdserver/etcdserverpb/rpc.proto [hunk 1]
```diff
@@ -527,10 +528,14 @@ message Compare {
...
     bytes value = 7;
+    // lease is the lease id of the given key.
+    int64 lease = 8;
+    // leave room for more target_union field tags, jump to 64
   }
+
   // range_end compares the given target to all keys in the range [key, range_end).
   // See RangeRequest for more details on key ranges.
-  bytes range_end = 8;
+  bytes range_end = 64;
```
Comment (xiang90) [inline etcdserver/etcdserverpb/rpc.proto]: this field is not in any release yet, right?

**Comment by xiang90** [comment]
do we also need to update etcdctl?

File: etcdserver/etcdserverpb/rpc.proto (hunk 1 above)
Comment (heyitsanthony) [inline etcdserver/etcdserverpb/rpc.proto]: right, caught it early

**Comment by heyitsanthony** [comment]
OK added etcdctl support

**Comment by xiang90** [comment]
lgtm

**Pull request merged**